### ISSUE TITLE
Joplin stops getting information from any external editors

### ISSUE THREAD

**Issue by saarantras** [laurent22/joplin#3536]
I open a note in an external editor, edit and save, edit and save for a while, then new edits in the external editor will suddenly stop appearing in Joplin (even after saving). If I close and re-open the external editor, the problem persists (changes in the editor are not reflected in Joplin) but **only for that note**. Other notes can be opened in an external editor, and edited and saved, and they will update in Joplin as expected. The only way to fix the problem is to restart Joplin.

I really like this program, but this bug renders it nigh-unusable for me.

(I have tried `gedit` with and without `--new-window` and `pluma`)

## Environment

Joplin version: 1.0.227 Revision: 4913fdb (master)
Platform: Linux
OS specifics: Linux Mint 18.3 Sylvia

## Steps to reproduce

1. Set up an external editor (gedit)
2. Open a note, edit and save, edit and save for a few minutes to hours
3. Check Joplin periodically, until the note stops updating.

## Describe what you expected to happen

I expected edits to save to Joplin.

## Logfile

![Screenshot_2020-07-20_13-21-21](https://user-images.githubusercontent.com/31054972/87983247-edad2100-ca8c-11ea-93d8-e9311009baf0.png)

**Comment by laurent22** [comment]
Did the bug happened when you took a screenshot of the log above?

**Comment by saarantras** [comment]
> Did the bug happened when you took a screenshot of the log above?

Yes. I started Joplin with the console open, used it until the bug occurred, then screnshotted the log and inserted it above. Let me know if you need any additional information.

Thank you for developing Joplin! I really like this program (other than this bug).

**Comment by laurent22** [comment]
Could you provide the portion of log.txt at the time this bug happens? This log would have detailed info about the external editor events, so it might help make sense of the bug. Also please write down the note ID that you were editing. https://joplinapp.org/debugging

**Comment by saarantras** [comment]
Log is attached. I tried to chop it down to just the relevant portion, but it is still very long. Let me know if there is anything missing you need.

Command to start external editor was `gedit --new-window` as can be seen in the logs.
[log-upload.txt](https://github.com/laurent22/joplin/files/4956755/log-upload.txt)

From the screenshot, the ID should be `0c01b576fe2342088240ffce7efda030` as you can see this was the one I was editing right before the image was taken.

**Comment by laurent22** [comment]
The file I would need is actually log.txt, but I think you've sent log-database.txt

**Comment by saarantras** [comment]
> The file I would need is actually log.txt, but I think you've sent log-database.txt

Whoops! Relevant log [log.txt](https://github.com/laurent22/joplin/files/4968074/log.txt)

I can also replicate the bug on a clean joplin install (no config, just one note) and send you those logs.

**Comment by laurent22** [comment]
If you could replicate on a clean install and provide the log that would be even better. If you could write down at what time happens the bug so that we can find the issue in the log that would help too.

**Comment by justwholegrain** [comment]
I just installed Joplin (news user here:). I have not tested an external editor, but for me this log (like in first post) appears every time I start Joplin. I found this when I was looking for a solution to the moving note problem (#3482). Maybe my logs will help. 

## Environment

Joplin version: 1.0.227 and Pre-release 1.0.231
Platform: Windows
OS specifics: Windows 7 64bit

## Logfile

[Console log when starting app.txt](https://github.com/laurent22/joplin/files/4977280/Console.log.when.starting.app.txt)
[log when starting app.txt](https://github.com/laurent22/joplin/files/4977281/log.when.starting.app.txt)

**Comment by saarantras** [comment]
I was unable to reproduce the bug in a clean install after several hours of trying, but the bug is intermittent, so I may have just had bad luck. I can try again later.

Meanwhile, here are two more examples of the bug happening in my normal "unclean" install. I have cropped the logs way down to just the exact relevant portion.

Example 1: 
Bug noticed : 2020-7-28 ~10:40 AM
Note id:  07c718b27a754817b8c0f024f7965640
Note Title : Week 4 : Arboleda (2020) The Free Market Can’t Prevent Latin America’s Coming Food Crisis.pdf
Last line that appeared in joplin before joplin stopped syncing from the external editor : - Anti-hunger protests in Chile
RELEVANT LOG : [log_err.txt](https://github.com/laurent22/joplin/files/4997429/log_err.txt)

Example 2:
Bug noticed : 2020-07-28 ~1:44 PM
Note id: 4ff3a712d89344cd8f319008ab3831ba
Note Title: Week 4 : 2020-7-28 lecture
Last line that appeared in joplin before joplin stopped syncing from the external editor : Sometimes it is paid for, sometimes it is simply seized.
RELEVANT LOG :  [log_err.txt](https://github.com/laurent22/joplin/files/4997468/log_err.txt)
